addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/internal/SmappeeService.java

```diff
@@ -0,0 +1,461 @@
+            logger.warn("Could not get access token");
+            return;
+        }
+
+        postMethod.addRequestHeader("Authorization", "Bearer " + accessTokenToInclude); // add the authorization header
+                                                                                        // to the request
+
+        try {
+            int statusCode = getClient.executeMethod(postMethod);
+            if (statusCode != HttpStatus.SC_OK) {
+                logger.warn("Post method failed: {}", postMethod.getStatusLine());
+                return;
+            }
+
+            return;
+        } catch (HttpException e) {
+            logger.warn("Fatal protocol violation: {}", e.toString());
+            return;
+        } catch (IOException e) {
+            logger.warn("Fatal transport error: {}", e.toString());
+            return;
+        } finally {
+            postMethod.releaseConnection();
+        }
+    }
+
+    private String GetAccessToken() {
+        if (accessToken != null && !accessToken.isEmpty() && accessTokenValidity != null
+                && accessTokenValidity.isBeforeNow()) {
+            return accessToken;
+        }
+
+        HttpClient postClient = new HttpClient();
+
+        if (accessTokenValidity != null) {
+            // get new accesstoken by using the refreshToken
+            PostMethod postMethod = new PostMethod("https://app1pub.smappee.net/dev/v1/oauth2/token");
+            postMethod.addRequestHeader("Accept", "application/json");
+
+            postMethod.addParameter("grant_type", "refresh_token");
+            postMethod.addParameter("refresh_token", refreshToken);
+            postMethod.addParameter("client_id", clientId);
+            postMethod.addParameter("client_secret", clientSecret);
+
+            try {
+                int statusCode = postClient.executeMethod(postMethod);
+                if (statusCode != HttpStatus.SC_OK) {
+                    logger.warn("Get Access Token failed: {}", postMethod.getStatusLine());
+                    return "";
+                }
+
+                String result = IOUtils.toString(postMethod.getResponseBodyAsStream());
+                SmappeeAccessTokenResponse accessTokenResponse = gson.fromJson(result,
+                        SmappeeAccessTokenResponse.class);
+
+                accessToken = accessTokenResponse.access_token;
+                refreshToken = accessTokenResponse.refresh_token;
+                accessTokenValidity = DateTime.now().plusSeconds(accessTokenResponse.expires_in);
+
+                return accessToken;
+
+            } catch (HttpException e) {
+                logger.warn("GetAccessToken on credentials : Fatal protocol violation: {}", e.toString());
+                return "";
+            } catch (IOException e) {
+                logger.warn("GetAccessToken on credentials : Fatal transport error: {}", e.toString());
+                return "";
+            } catch (Exception e) {
```

Comment:
ok,  
the catch(Exception) should not occur, except when smappee changes it's logic. if so, it should be logged with a specific message.

Reply:
If it would trigger when the logic changes, it can only be a subclass of RuntimeException because all other exceptions are checked, meaning they are known at compile time if they can be thrown at all. If you want to catch changed logic write it to catch RuntimeException.

Reply:
ok gson.fromJson might trigger a RuntimeException of their logic changes, so changed Exception -> RuntimeException (on 2 locations in getAccessToken)